activation/poet.go

```diff
@@ -538,7 +548,19 @@ func (c *poetService) Submit(
 		log.ZShortStringer("smesherID", nodeID),
 	)
 
-	// Try obtain a certificate
+	if err := c.verifyPhaseShiftConfiguration(ctx); err != nil {
+		if errors.Is(err, ErrIncompatiblePhaseShift) {
+			logger.Panic("failed to submit challenge",
+				zap.Error(err),
+				zap.String("poet", c.client.Address()))
+		}
+		logger.Warn("failed to submit challenge: couldn't fetch poet phase shift",
+			zap.Error(err),
+			zap.String("poet", c.client.Address()))
+		return nil, err
+	}
```

Comment:
NIT: same as above and the caller already logs the error returned here - no need to log it twice 🙂 

```suggestion
	err := c.verifyPhaseShiftConfiguration(ctx)
	switch {
	case errors.Is(err, ErrIncompatiblePhaseShift):
		logger.Panic("failed to submit challenge: incompatible phase shift",
			zap.String("poet", c.client.Address()),
		)
	case err != nil:
		return nil, err
	}
```